_stbt/keyboard.py

```diff
@@ -770,6 +805,86 @@ def wait_for_transition_to_end(
                                                stable_secs=stable_secs)
 
 
+@dataclasses.dataclass
+class _MutRegion:
+    x: int
+    y: int
+    right: int
+    bottom: int
+
+
+def _merge_regions(posns: list[_MutRegion]):
+    """Given a list of regions, merge any that are touching into larger regions.
+
+    This can currently only cope with rectangular regions.
+    """
+    if len(posns) <= 1:
+        # Common case
+        return posns
+    # First do horizontal merge:
+    out: list[_MutRegion] = []
+    last = None
+    for posn in posns:
+        if last and posn.x == last.right and posn.y == last.y:
+            last.right += 1
+        else:
+            last = posn
+            out.append(last)
+
+    # Now do vertical merge in >x^2 time because I'm lazy:
+    modified = True
+    while modified:
+        modified = False
+        for n, a in enumerate(out):
+            for m, b in enumerate(out[n + 1:], n + 1):
+                if a.bottom != b.y:
+                    continue
+                if a.x == b.x and a.right == b.right:
+                    a.bottom = b.bottom
+                    del out[m]
+                    modified = True
+                    break
+                # Only support rectangles for now:
+                if b.x < a.right and a.x < b.right:
+                    # Overlapping, but not equal
+                    raise NotImplementedError(
```

Comment:
ValueError perhaps?

Reply:
I see `ValueError` like an HTTP 4xx error - there's something wrong with the value that the user has passed.  This is more like a TODO: the user asked for something reasonable, but we just haven't got round to implementing it yet.  Maybe `NotImplementedError` isn't appropriate for that, but that's the reasoning anyway.

Maybe we need a `ToDoError` that points users at an issue in our GitHub so they can voice that they actually need this functionality.

Reply:
IOW the meaning in this case is "not implemented yet" error rather than "not implementable" error.

Also implied: I would consider a future change such that this does not raise `NotImplementedError` to be backwards compatible.